### PR TITLE
Fix the staker test

### PR DESCRIPTION
**Pull request by ImJeremyHe**
Close: #117 

### This PR:
- Fix the batch poster where we pulled the wrong `block_justification`
- Fix the WASM binary which didn't verify the block merkle proof when the txes is not found in the hotshot block
- Fix the tests regarding the updates in the light client contract

Relevent PRs:
- https://github.com/EspressoSystems/espresso-sequencer/pull/1625
- https://github.com/EspressoSystems/nitro-contracts/pull/8

### How to test this PR: 
Run this command locally since we don't check the staker test in the CI:
```
E2E_CHECK_STAKER=1 go test -v -count=1 -timeout 25m -run ^TestEspressoE2E$ github.com/offchainlabs/nitro/system_tests
```
There is a probablity that this test can not pass because of unknown issues for me and I guess it is because of limited resources in my computer.

### CODE REVIEW

File: cmd/replay/main.go
Comment (ImJeremyHe) [inline cmd/replay/main.go:1]: After doing this change, the e2e test fails. From my understanding, even there is no tx in the hotshot block, we should validate the block merkle proof, and that's why I made this. Not sure what's wrong with the validation pipeline

Comment (nomaxg) [reply cmd/replay/main.go:1]: Where is it failing exactly?

Comment (ImJeremyHe) [reply cmd/replay/main.go:1]: Failed to verify the merkle proof in the case that the hotshot block contains no txes

**Review by nomaxg — APPROVED** [review]
LGTM

**Pull request merged**